tests/compile-fail/needless_update.rs

```diff
@@ -0,0 +1,14 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![deny(needless_update)]
+
+struct S {
+    pub a: i32,
+}
+
+fn main() {
+    let base = S { a: 0 };
+    S { ..base }; // no error
+    S { a: 1, ..base }; //~ERROR update has no effect
+}
```

Comment:
Nit: I'd like to see a test case with more than one field and 0, 1, all fields updated.